gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePlugin.java

```diff
@@ -0,0 +1,261 @@
+            "com.palantir.baseline-error-prone.disable");
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().withPlugin("java", unused -> {
+            applyToJavaProject(project);
+        });
+    }
+
+    private void applyToJavaProject(Project project) {
+        project.getPluginManager().apply(ErrorPronePlugin.class);
+
+        SuppressibleErrorProneExtension extension =
+                project.getExtensions().create("suppressibleErrorProne", SuppressibleErrorProneExtension.class);
+
+        String version = Optional.ofNullable((String) project.findProperty("suppressibleErrorProneVersion"))
+                .or(() -> Optional.ofNullable(
+                        SuppressibleErrorPronePlugin.class.getPackage().getImplementationVersion()))
+                .orElseThrow(
+                        () -> new RuntimeException("SuppressibleErrorPronePlugin implementation version not found"));
+
+        // When auto-suppressing, there are two stages. The first runs a bytecode patched version of errorprone (via an
+        // artifact transform) that intercepts every error from every check and adds a custom fix, a
+        // @RepeatableSuppressWarnings annotation to the relevant statement/method/field/class. The second stage runs
+        // an unpatched version of errorprone that applies a single errorprone check: SuppressWarningsCoalesce,
+        // which will combine all the @RepeatableSuppressWarnings and @SuppressWarnings annotations into one
+        // normal @SuppressWarnings annotation.
+        setupErrorProneArtifactTransform(project);
+
+        project.getConfigurations().named(ErrorPronePlugin.CONFIGURATION_NAME).configure(errorProneConfiguration -> {
+            // Required so that we can run the runtime parts of the errorprone patching in suppressing stage 1 and
+            // also the SuppressWarningsCoalesce errorprone in suppressing stage 2.
+            errorProneConfiguration
+                    .getDependencies()
+                    .add(project.getDependencies()
+                            .create("com.palantir.suppressible-error-prone:suppressible-error-prone:" + version));
+        });
+
+        project.getTasks().withType(JavaCompile.class).configureEach(javaCompile -> {
+            configureJavaCompile(project, javaCompile);
+
+            ((ExtensionAware) javaCompile.getOptions())
+                    .getExtensions()
+                    .configure(ErrorProneOptions.class, errorProneOptions -> {
+                        configureErrorProneOptions(project, extension, javaCompile, errorProneOptions);
+                    });
+        });
+
+        addAnnotationDependencyForSuppressingStage2(project, version);
+
+        if (isAnyKindOfPatching(project)) {
+            project.afterEvaluate(_ignored -> {
+                // To allow refactoring near usages of deprecated methods, even when -Xlint:deprecation is specified,
```

Comment:
I understand what this is doing, but what does "near usages" mean?

Reply:
I actually just [copied this from baseline](https://github.com/palantir/gradle-baseline/blob/4c71b30d2ba8a81cfb63cba8bd163207188c82a3/gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineErrorProne.java#L85-L100). I did actually change the comment though - it's not just refactoring of deprecated methods, it if there is any usage of deprecated methods - see this test:

https://github.com/palantir/suppressible-error-prone/blob/ee8fced1def59cbf7c37b414ae4c5fe4ea43e4f8/gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy#L406-L434